front-end/app/src/pages/login/Form.js

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import axios from "axios";
+import Navbar from "../../components/nav/Nav";
+import HomePage from "./../home/Home";
+import BookStandPage from "./../book-stand/BookStand";
+import AboutPage from "./../about/About";
+import SettingsPage from "./../settings/Settings";
+import {BrowserRouter as Router, Switch, Route } from "react-router-dom";
+export default class Form extends React.Component {
+
+  state = {
+    username: "",
+    password: "",
+    errorMessage: "",
+    loggedIn: false,
+    onLogin: true,
+    onSignup: false
+  };
+
+  showSignUpPage = () => {
+    this.setState({
+      onSignup: true,
+      onLogin: false,
+      errorMessage: ""
+    });
+  }
+
+  showLoginPage = () => {
+    this.setState({
+      onLogin: true,
+      onSignup: false,
+      errorMessage: ""
+    });
+  }
+
+  /**
+   * TODO: implement more user friendly messages later
+   * This function handles displaying an error to the user if
+   * their login/signup credentials failed
+   * @param {Error} error - error response from failed
+   * user login/signup
+   */
+  handleError = (error) => {
+    const errorMessage = error.response.data;
+    this.setState({
+      errorMessage: errorMessage
+    });
+  }
+
+  /**
+   * This function handles bringing the user to the homepage
+   * if they've successfully logged in or signed up
+   */
+  handleSuccess = () => {
+    this.setState({
+      errorMessage: "",
+      loggedIn: true
+    });
+  }
+
+  // make call to API to validate the username and password
+  validate = async () => {
+
+    const usernameInput = this.state.username;
+    const passwordInput = this.state.password;
+
+    // update the states missing credentials and invalid credentials
+    // according to the API response
+    const BASE_URL = "http://localhost:8000";
```

Comment:
I think Elliot made a note that the base URL might need to change from localhost to something else? Check with him if this BASE_URL is fine for now or if should be different now,

Reply:
Still working on hosting the web-server. It will change then.